File: sms-backend/models/locker.go
package models

import "time"

// LockerFile represents a file stored in a student's Digital Locker (FE-12, FE-13, FE-14)
type LockerFile struct {
	ID         uint      `gorm:"primaryKey" json:"id"`
	StudentID  uint      `gorm:"not null" json:"student_id"`
	Student    Student   `gorm:"foreignKey:StudentID" json:"student,omitempty"`
	FileName   string    `gorm:"not null" json:"file_name"`
	FilePath   string    `gorm:"not null" json:"-"`              // internal path, never exposed
	FileSize   int64     `json:"file_size"`                      // bytes
	FileType   string    `json:"file_type"`                      // "pdf", "jpg", etc.
	Category   string    `json:"category"`                       // "Certificate", "Assignment", "Portfolio"
	IsPublic   bool      `gorm:"default:false" json:"is_public"` // teacher access (FE-14)
	UploadedAt time.Time `json:"uploaded_at"`
	CreatedAt  time.Time `json:"created_at"`
}

File: sms-frontend/src/types/locker.ts
import type { Student } from './academic'

export interface LockerFile {
  id: number
  student_id: number
  student?: Student
  file_name: string
  file_size: number
  file_type: string
  category: 'Certificate' | 'Assignment' | 'Portfolio'
  is_public: boolean
  uploaded_at: string
  created_at: string
}